msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java

```diff
@@ -102,9 +102,10 @@ private void initClientAuthentication(IClientCredential clientCredential) {
             this.clientCertAuthentication = true;
             this.clientCertificate = (ClientCertificate) clientCredential;
             if (Authority.detectAuthorityType(this.authenticationAuthority.canonicalAuthorityUrl()) == AuthorityType.ADFS) {
-                clientAuthentication = buildValidClientCertificateAuthoritySha1();
+                //When this was added, ADFS did not support SHA256 hashes for client certificates
+                clientAuthentication = buildValidClientCertificateAuthority(true);
             } else  {
-                clientAuthentication = buildValidClientCertificateAuthority();
+                clientAuthentication = buildValidClientCertificateAuthority(false);
```

Comment:
You could eliminate the if statement and call `buildValidClientCertificateAuthority` only once:

```
const useSha1 = (Authority.detectAuthorityType(this.authenticationAuthority.canonicalAuthorityUrl()) == AuthorityType.ADFS);
clientAuthentication = buildValidClientCertificateAuthority(useSha1)
```

Reply:
Good point, in the latest commit I moved this logic down into the `buildValidClientCertificateAuthority` method. Now that method does the ADFS authority check itself, with a comment explaining why it's done.